src/openforms/authentication/utils.py

```diff
@@ -44,3 +46,10 @@ def store_registrator_details(
     RegistratorInfo.objects.update_or_create(
         submission=submission, defaults=registrator_auth
     )
+
+
+def is_authenticated_with_plugin(request: Request, expected_plugin: str) -> bool:
```

Comment:
I can't shake the feeling that we have more/similar checks that would benefit from refactoring to this utility, e.g. the submission resume flow. Have you checked those?